### DISCUSSION TITLE
Why not use archive.apache.org instead of dlcdn.apache.org to download Maven ?

### DISCUSSION THREAD

**Discussion by jonathan-schiltz-cap4lab** [actions/virtual-environments#7449] [Image configuration]
Hello ! In [https://github.com/actions/runner-images/blob/main/images/linux/scripts/installers/java-tools.sh line 131](https://github.com/actions/runner-images/blob/0f2310415608223eb4141ecdbd1101d307f26b06/images/linux/scripts/installers/java-tools.sh#L131)
`mavenDownloadUrl="https://dlcdn.apache.org/maven/maven-3/${mavenVersion}/binaries/apache-maven-${mavenVersion}-bin.zip"` We use `dlcdn.apache.org` to download the Maven binaries. However, `dlcdn` hosts only latest patch of every minor. You're currently using 3.8.8 (hardcoded in a config file). So when 3.8.9 is released, the image build will fail to fetch 3.8.8.

Why don't we use `archive.apache.org` instead as the url will "forever" serve the version ? Maintainers would not have to rush the Maven upgrade when Apache releases a patch.

The line linked above would become `mavenDownloadUrl="https://archive.apache.org/dist/maven/maven-3/${mavenVersion}/binaries/apache-maven-${mavenVersion}-bin.zip"`. Please mind the `/dist` between the `archive.apache.org` and the `/maven`.

Thanks for your consideration and all the efforts you make to deliver these images to the community.
I hope this idea will be useful to the maintainers.

Best regards,
Jonathan

**Comment by v-sergei-pyshnoi** [comment]
HI @jonathan-schiltz-cap4lab . Thanks for your advice! We will take look.
